### DISCUSSION TITLE
PopoverButton use with `:as` prop produces weird behavior

### DISCUSSION THREAD

**Discussion by briancbarrow** [tailwindlabs/headlessui#2562] [Help]
https://github.com/tailwindlabs/headlessui/assets/13695093/7de5d423-cc3c-4845-98cd-addc079c847b

As you can see in the screen recording above, when we use the `:as` prop on the `PopoverButton` component we get behavior where the popover doesn't close when the button is clicked again like it does when that prop is not present. 

Currently we are using a custom `Button` component, but even when the component we use is reduced to just the following template we still see the issue. 

```vue
<template>
    <button
        type="button"
        class="relative rounded-md border shadow-sm font-medium leading-tight
                disabled:cursor-not-allowed active:enabled:shadow-inner active:enabled:translate-y-px
                focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 px-4 py-2 text-sm
                bg-white text-gray-700 border-gray-300 hover:bg-gray-50 active:bg-gray-100 disabled:bg-gray-50
                disabled:text-gray-500 disabled:border-gray-200"
    >
        Test
    </button>
</template>
```

Here is a recording of what we would expect the behavior to be:

https://github.com/tailwindlabs/headlessui/assets/13695093/e6415239-8043-450c-ac0e-b35c8570076d

Are we just using the `:as` prop wrong? 

Edit: It does work fine if we do not use a button as the tag. So this works:
```vue
<template>
    <div
        class="relative rounded-md border shadow-sm font-medium leading-tight
                disabled:cursor-not-allowed active:enabled:shadow-inner active:enabled:translate-y-px
                focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 px-4 py-2 text-sm
                bg-white text-gray-700 border-gray-300 hover:bg-gray-50 active:bg-gray-100 disabled:bg-gray-50
                disabled:text-gray-500 disabled:border-gray-200"
    >
        Test
    </div>
</template>
```

Is the `button` tag emitting an event that is messing with the Popover state?

**Comment by Megarsh** [comment]
+1 for visibility, almost exact same issue on my end.

Here's a quick reproduction for the issue - [https://codesandbox.io/s/currying-hooks-xwt7gs?file=/src/App.vue](https://codesandbox.io/s/currying-hooks-xwt7gs?file=/src/App.vue)

The issue appeared on the newest 1.7.14 version, tested it on 1.7.12 and it worked as expected
